frontend/src/__tests__/cypress/cypress/support/commands/application.ts

```diff
@@ -148,31 +152,34 @@ declare global {
   }
 }
 
-Cypress.Commands.add('visitWithLogin', (url, user = HTPASSWD_CLUSTER_ADMIN_USER) => {
+Cypress.Commands.add('visitWithLogin', (relativeUrl, credentials = HTPASSWD_CLUSTER_ADMIN_USER) => {
   if (Cypress.env('MOCK')) {
-    cy.visit(url);
+    cy.visit(relativeUrl);
   } else {
-    cy.intercept('GET', url, { log: false }).as('visitWithLogin');
-
-    cy.visit(url, { failOnStatusCode: false });
-
-    cy.wait('@visitWithLogin', { log: false }).then((interception) => {
-      if (interception.response?.statusCode === 403) {
-        cy.log('Do login');
-        // do login
+    let fullUrl: string;
+    if (relativeUrl) {
+      fullUrl = new URL(relativeUrl, Cypress.config('baseUrl') || '').href;
+    } else {
+      fullUrl = new URL(Cypress.config('baseUrl') || '').href;
+    }
+    cy.step(`Go to: ${fullUrl}`);
+    cy.intercept('GET', fullUrl, { log: false }).as('page');
+    cy.visit(fullUrl, { failOnStatusCode: false });
+    cy.wait('@page', { log: false }).then((interception) => {
+      const statusCode = interception.response?.statusCode;
+      if (statusCode === 403) {
+        cy.log('Log in');
         cy.get('form[action="/oauth/start"]').submit();
-        cy.findAllByRole('link', user.AUTH_TYPE ? { name: user.AUTH_TYPE } : {})
+        cy.findAllByRole('link', credentials.AUTH_TYPE ? { name: credentials.AUTH_TYPE } : {})
           .last()
-          .click();
-        cy.get('input[name=username]').type(user.USERNAME);
-        cy.get('input[name=password]').type(user.PASSWORD);
+          .then(($link) => {
+            cy.wrap($link).click();
+          });
+        cy.get('input[name=username]').invoke('val', '').type(credentials.USERNAME);
+        cy.get('input[name=password]').invoke('val', '').type(credentials.PASSWORD);
```

Comment:
Can we use `fill` instead?
```suggestion
        cy.get('input[name=username]').fill(credentials.USERNAME);
        cy.get('input[name=password]').fill(credentials.PASSWORD);
```